en/middleware/mavlink.md

```diff
@@ -519,3 +525,55 @@ However if you use your own dialect then it should include ArduPilotMega.xml (or
 ### Updating MAVSDK
 
 See the MAVSDK docs for information about how to work with [MAVLink headers and dialects](https://mavsdk.mavlink.io/main/en/cpp/guide/build.html).
+
+### Updating MAVProxy
+
+[MAVProxy](https://ardupilot.org/mavproxy/) is UAV ground station software package for MAVLink based systems written in Python.
+While not optimized for PX4 or widely promoted in the PX4 community, we mention it here because it is one of the easiest ground stations to update with custom messages, and is therefore one of the easiest ways to test that your custom messages are being sent.
+
+[MAVProxy](https://ardupilot.org/mavproxy/) depends on [Pymavlink](https://mavlink.io/en/mavgen_python/).
+The approach described below is to install MAVProxy, remove the default-installed Pymavlink, and then rebuild Pymavlink with the new messages.
+
+The steps are:
+
+1. Install MAVproxy using the command:
+
+   ```sh
+   python -m pip install mavproxy
+   ```
+
+1. Remove the Pymavlink package installed with MAVProxy:
+
+   ```sh
+   python -m pip uninstall pymavlink
+   ```
+
+1. Clone the MAVLink repository (which also contains Pymavlink)
+
+   ```sh
+   git clone https://github.com/mavlink/mavlink.git --recursive
+   cd mavlink
+   ```
+
+1. Copy your custom message XML into the [message_definitions/v1.0](https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0) folder
+1. Generate the headers:
+
+   ```sh
+   python3 -m pymavlink.tools.mavgen --lang=C --wire-protocol=2.0 --output=generated message_definitions/v1.0/your_dialect.xml
```

Comment:
@wasaybaig Are you sure you need this step?
This builds the C headers in mavlink/mavlink, not Python headers so I don't think this would be used.

Reply:
@hamishwillee it works this way also but we can change the language to Python. But it is important to generate the headers with your new message or MAVProxy will not be able to understand the message.

Reply:
But MAVProxy uses Python so this is a null operation - you haven't built anything that MAVProxy can use.

Reply:
I guess I'll have to try it.

Reply:
This is not needed. setup.py pulls any xml files in the parent definitions folder and builds them. Tested.